### ISSUE TITLE
CLI exits with 1 on warnings even without `--warnings`

### ISSUE THREAD

**Issue by disrupted** [DetachHead/basedpyright#1113]
### Description

The [CLI documentation](https://docs.basedpyright.com/latest/configuration/command-line/#command-line) says that it will use exit code 1 if warnings are reported with the `--warnings` flag. But it does the same without the flag.

```
0 errors, 227 warnings, 0 notes
❯ echo $?
1
```

**Comment by disrupted** [comment]
It seems that you need to set `failOnWarnings=false` in the config (in my case pyproject.toml)

**Comment by DetachHead** [comment]
this is because `failOnWarnings` is enabled by default in with `typeCheckingMode = "recommended"`. [more info](https://docs.basedpyright.com/latest/configuration/config-files/#recommended-and-all). 

i should definitely update the cli documentation to mention this though
